largestPrimeFactor.js

```diff
@@ -1,6 +1,23 @@
 exports.largestPrimeFactor = function(n){
   var primeNumber = 0;
-  // do your work here
-
+  factor = 0;
```

Comment:
So I pulled this down and ran the test but it doesn't look like it works or passes any of the tests. Also, the deeply nested structure makes this very hard to read.